src/Java.Runtime.Environment/Java.Interop/JreRuntime.cs

```diff
@@ -87,12 +80,22 @@ static unsafe JreRuntimeOptions CreateJreVM (JreRuntimeOptions builder)
 			if (builder == null)
 				throw new ArgumentNullException ("builder");
 
+			bool onMono = Type.GetType ("Mono.Runtime", throwOnError: false) != null;
+			if (onMono) {
+				builder.ValueManager            = builder.ValueManager              ?? new MonoRuntimeValueManager ();
+				builder.ObjectReferenceManager  = builder.ObjectReferenceManager    ?? new MonoRuntimeObjectReferenceManager ();
+			}
+			else {
+				builder.ValueManager            = builder.ValueManager              ?? new ManagedValueManager ();
+				builder.ObjectReferenceManager  = builder.ObjectReferenceManager    ?? new ManagedObjectReferenceManager (builder.JniGlobalReferenceLogWriter, builder.JniLocalReferenceLogWriter);
+			}
+
```

Comment:
Nice. We don't have to create the managers before creating the VM. I like that change.